src/utils/ConvertToEmbeddableUrl.tsx

```diff
@@ -8,28 +15,55 @@ export interface ConvertToEmbeddableUrlOptions {
 // @debt Replace the naive logic with use of new URL(url) which provides standard parsing before doing checks
 export const convertToEmbeddableUrl: (
   options: ConvertToEmbeddableUrlOptions
-) => string = ({ url, autoPlay = DEFAULT_VENUE_AUTOPLAY }) => {
-  if (url?.includes("youtube")) {
-    url = url?.replace("watch?v=", "embed/");
-  } else if (
-    url?.includes("vimeo") &&
-    !url?.includes("player") &&
-    // NOTE: If you have a scheduled live event, it gives you a different embed code
-    !url?.includes("vimeo.com/event")
+) => string = ({ url, autoPlay }) => {
+  if (!url) {
+    return "";
+  }
+
+  let urlObj = new URL(url);
+  const { host, searchParams, pathname } = urlObj;
+  const youtubeVideoId = searchParams.get("v") || pathname.slice(1);
+  const twitchVideoId = searchParams.get("video");
+  const twitchId = searchParams.get("channel") || twitchVideoId;
+  const isTwitch = host.includes(TWITCH_SHORT_URL);
+
```

Comment:
Moving 
```ts
  if (autoPlay) {
    const autoPlayValue = isTwitch ? "true" : "1";
    url.searchParams.set("autoplay", autoPlayValue);
  } else {
    url.searchParams.delete("autoplay");
  }
```
before the other checks will help reducing the `if-else-if` nesting by fail fast i.e. just `return url;` inside the `if`

Reply:
Better yet, extract a helper function, then specialized delegate functions per case:

```ts

const withAutoPlay = (url: URL, autoPlay: "1" | "true" | false | undefined) => {
  if (autoPlay) {
    url.searchParams.set("autoplay", autoPlay);
  }
  return url.href;
};

type ConvertTwitchUrlOptions = {
  pathname: string;
  twitchId?: string;
  twitchVideoId?: string;
  autoPlay?: boolean;
};
const convertTwitchUrl: (options: ConvertTwitchUrlOptions) => string = ({
  pathname,
  twitchId,
  twitchVideoId,
  autoPlay,
}) => {
  const id = pathname.split("/")?.slice(-1)?.[0] || twitchId || "";
  const twitchUrl = new URL(TWITCH_EMBED_URL);

  if (pathname.includes("video") || twitchVideoId) {
    twitchUrl.searchParams.set("video", id);
  } else {
    twitchUrl.searchParams.set("channel", id);
  }

  twitchUrl.searchParams.set("parent", "localhost");

  return withAutoPlay(twitchUrl, autoPlay && "true");
};```

This way `convertToEmbeddableUrl()` becomes a facade that just checks which service it is called about and returns the results from the helper functions

Reply:
thanks, fixed in ea1c3b0